book/special.qmd

```diff
@@ -851,188 +851,188 @@ The experiment shows that using the K-means algorithm with four centers has the
+% learner = lrn("classif.rpart", predict_type = "prob")
+% rsmp_nsp = rsmp("repeated_cv", folds = 3, repeats = 2, id = "NSpCV")
+% rsmp_sp = rsmp("repeated_spcv_coords", folds = 3, repeats = 2, id = "SpCV")
+%
+% design = benchmark_grid(task, learner, c(rsmp_nsp, rsmp_sp))
+% bmr = benchmark(design)
+% bmr$aggregate(msr("classif.acc"))[, c(5, 7)]
+% ```
+%
+% In the above example, the non-spatial resampling method means that the train and test data are very similar due to spatial autocorrelation, and so this is almost as bad as evaluating a model on the training data (see @sec-basics).
+% In contrast, the spatial method has accommodated for autocorrelation and the test data is therefore less correlated (though still some association will remain) with the training data.
+% Visually this can be seen using built-in `r ref("autoplot")` methods.
+% In @fig-sprsmp we visualize how the task is partitioned according to the spatial resampling method (@fig-sprsmp top) and non-spatial resampling method (@fig-sprsmp bottom).
+% There is a clear separation between coordinates using the 'correct' spatial resampling method whereas the train and test splits overlap significantly (and are therefore highly correlated) using the non-spatial method.
+%
+% ```{r special-047}
+% #| fig-cap: Scatterplots show separation of train (blue) and test (orange) data for the first three (left to right) folds of the first repetition of the cross-validation. The top row is spatial resampling where train and test data are clearly separated. The bottom row is non-spatial resampling where there is overlap in train and test data.
+% #| fig-alt: Six scatter plots are shown with y-axes from 4 degrees South to 3.97 degrees South and x-axes from 79.08 degrees West to 79.06 degrees West. The top row of plots show a very clear separation with no overlap between blue and orange datapoints, whereas the bottom low shows significant overlap.
+% #| label: fig-sprsmp
+%
+% library(patchwork)
+% autoplot(rsmp_sp, task, fold_id = c(1:3), size = 0.7) /
+%   autoplot(rsmp_nsp, task, fold_id = c(1:3), size = 0.7) &
+%   scale_y_continuous(breaks = seq(-3.97, -4, -0.01)) &
+%   scale_x_continuous(breaks = seq(-79.06, -79.08, -0.02))
+% ```
+%
+% Now we have seen why spatial resampling methods matter we can take a look at what is implemented in `r mlr3spatiotempcv`.
+% At the time of writing we have implemented 19 different resampling methods for spatial and spatiotemporal analysis.
+% These are categorized into:
+%
+% * Blocking -- Create rectangular blocks in 2D/3D space (as opposed to partitioning in 1D space used in non-spatial methods)
+% * Buffering -- Create buffering zones to remove observations between train and test sets
+% * Spatiotemporal clustering -- Clusters based on coordinates (and/or time-points)
+% * Feature space clustering -- Clusters based on feature space and not necessarily spatiotemporal
+% * Partitioning -- Grouped by factor variables
```

Comment:
This description would optimally align with the categorization we did in https://arxiv.org/abs/2110.12674 where we used "Custom (partitioning)" for the factor var grouping and also added more information about each category.

Also partly available at https://mlr3spatiotempcv.mlr-org.com/articles/mlr3spatiotempcv.html#upstream-packages-and-scientific-references.